### PR TITLE
Play By Play Widget

### PR DESCRIPTION
**Pull request by ThierryBC24**
Add the play by play widget and fix the missing playoff data.

### CODE REVIEW

File: ift6758/data/play_by_play_data.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        :param season: Season (ex: 2022 for the 2022-2023 season)
+        :return: DataFrame
+        """
+        file_path = os.path.join(self.base_path, f"{season}_play_by_play.csv")
+
+        # If a local file exists, load the data from the file
+        if os.path.exists(file_path):
+            print(f"Loading local data for the season {season}-{season+1}...")
+            return pd.read_csv(file_path)
```
Comment (RexBergy) [inline ift6758/data/play_by_play_data.py:28]: read_json

File: ift6758/data/play_by_play_data.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            os.makedirs(self.base_path)
+
+    def get_data(self, season: int) -> pd.DataFrame:
+        """
+        Fetch data for a specific season, either from a local save or from the NHL's API.
+
+        :param season: Season (ex: 2022 for the 2022-2023 season)
+        :return: DataFrame
+        """
+        file_path = os.path.join(self.base_path, f"{season}_play_by_play.csv")
```
Comment (RexBergy) [inline ift6758/data/play_by_play_data.py:23]: .json

File: ift6758/data/play_by_play_data.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        if os.path.exists(file_path):
+            print(f"Loading local data for the season {season}-{season+1}...")
+            return pd.read_csv(file_path)
+
+        # Else, download data from the API
+        print(f"Downloading data for the season {season}-{season+1}...")
+        data = self.fetch_season_data(season)
+
+        # Save data into a CSV file
+        data.to_csv(file_path, index=False)
```
Comment (RexBergy) [inline ift6758/data/play_by_play_data.py:35]: to_json

File: ift6758/data/play_by_play_data.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    def get_all_data(self) -> pd.DataFrame:
+        """
+        Fetch all collected data into the local save.
+
+        :return: DataFrame with all data
+        """
+        all_data = []
+        for file in os.listdir(self.base_path):
+            if file.endswith('.csv'):
```
Comment (RexBergy) [inline ift6758/data/play_by_play_data.py:76]: .json

File: ift6758/data/play_by_play_data.py (hunk 1 above)
Comment (momojope) [reply ift6758/data/play_by_play_data.py:28]: bien !!

**Review by momojope — APPROVED** [review]

**Pull request merged**